### ISSUE TITLE
Allow users to easily get re-published assigned surveys from within the app

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-product-design#222]
## Current situation

Currently you create an assignment on the dashboard that holds your published surveys. After opening the app the surveys (and forms) download automatically onto the phone and the user sees a small notification about this action. Then the surveys appear in the list in the menu. 

If the surveys are changed on the dashboard and republished, these changes need to be sent to the phone so the user works with the correct version of the form from the survey. In order to receive the latest version of the form, the user has to fully close the app (make sure it does not run in the background of the phone) and reopen it again. Many users do not know how to do this, or sometimes they have to repeat the process to get the last form version to come through.  

### Why? 
There is no clear and simple way to know if your survey version on the phone is the latest and the download steps for getting the re-pulished form in are not so clear to the users. 

### Who? 
Users of the Flow app. We tend to have two categories of Flow app users: 
- users handy with smartphones and Android apps > these are the daily Android users, who have their own smart phones, understand how apps work and do not have troubles conducting the tasks related to the Flow app. They also can quickly pick up the workflow without receiving training sessions. 
- users who have never (or seldom) worked with a smarthphone and the Flow app is their 1st experience. We tend to teach therm how to type on a phone and they need to see explicit indications and know the specific steps on how to use the app

## Opportunity 

- How can we make it easier, and more straightforward, for the user to get the republished surveys on their phones (the latest version of the form)? 
- Is there a way to notify the user that there is a new version of the form that he has on his device?

## Next Steps 

After thinking about the idea @muloem shared in the Flow chat, to have a "trigger" button for the users in the app Settings, I realised that there might be a bit more to it. I think we should consider and ideal solution for this issue and then see how we can break it down into small chunks (implementable tasks) and put those in the mobile repository. 

@valllllll2000 and @loicsans perhaps we can brainstorm (Story Map) a solution once there is time.

**Comment by valllllll2000** [comment]
Step 1: We can already do what @muloem suggested and add the possibility to check for survey updates in settings screen.

Step 2: actively notify the user about survey changes may also be possible.
This weekend I have been playing with FCM or Firebase Cloud Messaging and on mobile it is very easy to setup. FCM is the new GCM from google. Here is a link with server documentation https://firebase.google.com/docs/cloud-messaging/server. I do not know how hard it would be to include some kind of trigger in the backend after a survey has been modified to send and FCM notification to users.

**Comment by muloem** [comment]
Just one correction, when I mentioned having a way to check for survey updates, I did not imply that it should be in the `Settings` part of the app.  To me this is more a concern that belongs say in the *survey list* part of the app, or the main activity of the app, and we need to think whether such a button should be per survey or that it should update *all* surveys that have been republished, also whether such a button should be used as well to initiate the submission of data thats not yet been sent to the dashboard, as well as to update the list of data points on the device

**Comment by muloem** [comment]
@valllllll2000 we can easily start with the pull model for now, i.e the user actively requests an update or the update happens automatically when the app has been restarted.  The push notifications would need some more study to see how much they would/wouldn't tie us into the GAE platform.

**Comment by janagombitova** [comment]
@muloem sorry, I misunderstood your comment from the chat. I think we need to do some proper thinking before we change any user interaction and need to involve the design team as well on this one. It can be a nice case to try out User Story Mapping for us, as the problem is very specific. 

I agree we should start by ensuring the current steps users have to take work. So once the user reopens the app the most recent version of the survey are downloaded to the phone with all the changes https://github.com/akvo/akvo-flow-mobile/issues/471 and that after publishing a survey the version number gets updated correctly https://github.com/akvo/akvo-flow/issues/1403

**Comment by janagombitova** [comment]
Today we spoke with @valllllll2000 and @Kiarii briefly about this problem. We realised that there are a few different cases here: 
- assignment is created and Aplhonso wants to get all the forms from that assignment into the device
- form version of a form that is already on Alphonso's device is updated and he needs to have the latests version on the phone 
- another form has been added to the existing assignment and Aplhonso needs to get that form onto his device

Aplhonso can either close and re-open the app to get in the forms following the above cases. He can use the "Reload all surveys" (see thoughts around relocating this action to the survey list in the side menu https://github.com/akvo/akvo-flow-mobile/issues/636), but this only takes the surveys currently in the app and reloads them, so not all cases are covered by this action. 

How can we simplify his experience and give him a way to get the latests versions of the forms he has on his device, or the new forms assigned to his device?
